### ISSUE TITLE
Device becomes warm while the app is open

### ISSUE THREAD

**Issue by An-anonymous-coder** [VadimBoev/FlappyBird#31]
Pixel 8 running GrapheneOS (Android 14)

While using the app, the battery begins to heat up more than it should. This is an indication of nonessential loops or unoptimized code. 

Check to see if any code is running while the app is idle that is not required.

**Comment by VadimBoev** [comment]
The code is available, it can be viewed and analyzed. I'm closing this, it sounds weird.

**Comment by illerokcob** [comment]
No, he was right, I double checked on a rooted Android 11 device the cpu usage with top, and I found the following:
- When the app is open, cpu usage is around 50-60% on one core:
![Screenshot_20241005-143938_Trebuchet_1](https://github.com/user-attachments/assets/93adaf92-38c4-4f49-a9c4-195ad06db1e9)
- When the app is in the background, cpu usage is on maximum on one core, 100%:
![Screenshot_20241005-143919_Trebuchet_1](https://github.com/user-attachments/assets/4c2a88fd-17bb-453d-8757-d60e773b6c61)

My test can be replicated easily on a rooted emulator as well with adb.
I'd suggest reviewing the code in main.c and checking for state's activityState alongside destroyRequested because it can detect when the app is backgrounded.

**Comment by WSQS** [comment]
I think we should add a nanosleep in the loop of android_main

**Comment by VadimBoev** [comment]
> I think we should add a nanosleep in the loop of android_main

from the point of view of working with threads, this is a good idea, but I can't be sure that this is a good practice for android native
